app/controllers/paypal_accounts_controller.rb

```diff
@@ -214,7 +214,18 @@ def flash_error_and_redirect_to_settings(error_response: nil, error_msg: nil)
       end
 
     flash[:error] = error_msg
-    redirect_to new_paypal_account_settings_payment_path(@current_user.username)
+    redirect_to error_redirect_to
+  end
+
+  def error_redirect_to
```

Comment:
I'd have this return just the action. That way you would avoid the assign from conditional -pattern that is always a bit hard to read, and also an extra local variable.